Fixes from Juelich jureca deployment

pramodk opened the pull request:
- flex & bison dependency missing after mod2c was made submodule

> @jorblancoa : py-bluepy and related python packages are failing to build. (will discuss offline).

pramodk committed:
Fixes from Juelich jureca deployment
  - flex & bison dependency missing after mod2c was made submodule
  - use boost from external
  - bzip2 url changed, see #495

pramodk committed:
Fixes from Jureca deployment
  - serial prefix for neuron~mpi
  - coreneuron flex/bison dependency was missing
  - install numpy explicitly to avoid bluepy building issue with python2

pramodk committed:
enable libxslt as bluepy fails to load on booster partition

pramodk committed:
enable libxslt module as its required at runtime in LD_LIBRARY_PATH for python libraries

pramodk committed:
Merge branch 'develop' into pr/juelich-fixes

pramodk committed:
Merge branch 'pr/juelich-fixes' into pr/juelich-update

pramodk committed:
revert changes to spack modules.yml

pramodk committed:
Merge branch 'develop' into pr/juelich-update

pramodk commented:
ping @jorblancoa

pramodk committed:
Merge branch 'develop' into pr/juelich-update

Commit:
Add python 3 to system paths

matz-e commented on sysconfig/jureca-booster/modules.yaml:
```diff
@@ -27,6 +27,7 @@ modules:
           '+profile': 'profile'
           '^coreneuron+knl': 'knl'
           'neuron~mpi': 'serial'
+          '^python@3.6.8': 'python3'
```
Why?

jorblancoa replied:
We need it for
https://github.com/cnr-ibf-pa/hbp-bsp-issues/issues/525#issuecomment-590703254

Commit:
Remove / from module suffix (consistent with booster)

pramodk committed:
Add new neurodamus-core with 2.9.3

pramodk commented:
Please retest

Commit:
* Add intel-mpi as default mpi provider
 * Whitelist brion module
 * Set boost as buildable

Commit:
Add daint changes
 * Whitelist modules (mousify, py-sonata-network-reduction, brion)
 * Enable autoload modules

Commit:
Merge daint and julich branches
Update packages for daint

Commit:
Merge branch 'develop' into pr/juelich-update

Commit:
Revert pandas version dependency in sonata-network reduction

Commit:
Remove python3 suffix
Add python2 suffix for neuron
Whitelist neuron with intel
Whitelist py-* with gcc8
Use boost 1.70.0

Commit:
Use external python3 packages in Jureca and Piz-Daint

Commit:
Add py-numpy as external package

Commit:
Add deployment installation scripts in daint and jureca
Reduce dependency requirements of sonata-reduction for numpy, pandas and h5py packages in order to be used as external

Commit:
Whitelist neuron serial/parallel in daint

Commit:
Jureca changes
 - Whitelist neuron serial/parallel
 - Update external py-pandas version

Commit:
Deployment scripts in jureca and daint
 -Install bluepyopt and sonata-network-reduction with neuron+mpi

Commit:
Added coreneuron knl variant to deployment script (jureca)

Commit:
py-sonata-network-reduction0.0.6

matz-e committed:
zeromq: build with IntelⓇ in external packages.

matz-e committed:
zeromq: create IntelⓇ section in externals.

matz-e committed:
deploy: name external IntelⓇ section right.

Commit:
Merge remote-tracking branch 'aleksei/py-sonata-network-reduction0.0.6' into pr/juelich-update

pramodk committed:
Add configure step for py-zmq to explicitly pass zeromq directory
 - See https://github.com/zeromq/pyzmq/wiki/Building-and-Installing-PyZMQ#building-pyzmq-from-source
 - This is require because Spack doesn't set PKG_CONFIG and many
   other variables for external packages. This results in use of system package.

pramodk committed:
flake-8

pramodk committed:
Bump py-zmq to be able to re-build py-zmq

pramodk committed:
Merge branch 'deploy/external-intel-zmq' into pr/juelich-update

Commit:
Removing py-pandas and py-matplotlib from external

pramodk committed:
Add missing neuron%intel dependencies

Commit:
Remove python packages from external in Jureca
Update deployment scripts
Update intel build system to support intel-mpi-external

pramodk committed:
Bug fix for importing neuron+mpi compiled mechanism library
 - also added latest releases of bluepyopt and efel

pramodk committed:
Deploy latest efel as well

pramodk commented:
Please retest

pramodk commented on sysconfig/daint/deploy_simstack.sh:
```diff
@@ -0,0 +1,83 @@
+#spack install --dirty --keep-stage -v py-bluepyopt%gcc^neuron~binary~mpi ^python@2.7.15
+spack spec -I neuron %intel ^python@2.7.15 ^mpich
+spack install --dirty --keep-stage -v neuron %intel ^python@2.7.15 ^mpich
+
+spack spec -I neuron~mpi %intel ^python@2.7.15
```
```suggestion
```

pramodk commented on sysconfig/daint/deploy_simstack.sh:
```diff
@@ -0,0 +1,83 @@
+export PATH=$SPACK_ROOT/bin:$PATH
+source $SPACK_ROOT/share/spack/setup-env.sh
+
+#chmod -R ugo+w $DEPLOYMENT_HOME
```
```suggestion
```

pramodk commented on sysconfig/daint/deploy_simstack.sh:
```diff
@@ -0,0 +1,83 @@
+
+module swap PrgEnv-cray PrgEnv-intel
+module load daint-mc
+# PYTHON 2 packages
+#spack spec -Il neurodamus-hippocampus+coreneuron %intel ^python@2.7.15 ^synapsetool%gcc
+#spack install --keep-stage neurodamus-hippocampus+coreneuron %intel ^python@2.7.15 ^synapsetool%gcc
+#spack install --keep-stage neurodamus-neocortex+coreneuron %intel ^python@2.7.15 ^synapsetool%gcc
+#spack install --keep-stage neurodamus-mousify+coreneuron %intel ^python@2.7.15 ^synapsetool%gcc
+
+#spack spec -I py-bluepy%gcc ^python@2.7.15
+#spack install --dirty --keep-stage -v py-bluepy%gcc ^python@2.7.15
+#spack spec -I -l py-bluepyopt%gcc^neuron~binary~mpi ^python@2.7.15 ^py-tornado@4.4.0 ^py-ipykernel@4.5.0 ^py-ipython@5.1.0
+#spack install --dirty --keep-stage -v py-bluepyopt%gcc^neuron~binary~mpi ^python@2.7.15
```
```suggestion
```

pramodk commented on sysconfig/daint/deploy_simstack.sh:
```diff
@@ -0,0 +1,83 @@
+
+module swap PrgEnv-intel PrgEnv-gnu
+
+spack spec -I py-bluepy%gcc ^python@3.6.5
```
```suggestion
spack spec -Il py-bluepy%gcc ^python@3.6.5
```

pramodk commented on sysconfig/daint/deploy_simstack.sh:
```diff
@@ -0,0 +1,83 @@
+# Re-generate modules
+spack module tcl refresh --delete-tree -y
+cd $DEPLOYMENT_HOME/install/modules/tcl/cray-cnl7-haswell
+find py* -type f -print0|xargs -0 sed -i '/PYTHONPATH.*\/neuron-/d'
```
add comment here explaining why we do this.

pramodk commented on sysconfig/daint/deploy_simstack.sh:
```diff
@@ -0,0 +1,83 @@
+#ln -s $DEPLOYMENT_HOME/install/modules/tcl/cray-cnl6-haswell $DEPLOYMENT_HOME/modules
+#chmod -R ugo-w $DEPLOYMENT_HOME
+
+#cd $BASE_DIR
+#rm modules
+#ln -s $DEPLOYMENT_HOME/install/modules/tcl/cray-cnl6-haswell modules
```
I think can be removed? I am thinking of not making default symlinks anymore

jorblancoa replied:
I agree, removed!

pramodk commented on sysconfig/daint/deploy_simstack.sh:
```diff
@@ -0,0 +1,83 @@
+find py* -type f -print0|xargs -0 sed -i '/PYTHONPATH.*\/neuron-/d'
+find neuro* -type f -print0|xargs -0 sed -i '/module load mpich/d'
+
+#ln -s $DEPLOYMENT_HOME/install/modules/tcl/cray-cnl6-haswell $DEPLOYMENT_HOME/modules
+#chmod -R ugo-w $DEPLOYMENT_HOME
```
```suggestion
ln -s $DEPLOYMENT_HOME/install/modules/tcl/cray-cnl6-haswell $DEPLOYMENT_HOME/modules
chmod -R ugo-w $DEPLOYMENT_HOME
```

pramodk commented on sysconfig/jureca-booster/deploy_simstack.sh:
```diff
@@ -0,0 +1,98 @@
+fi
+
+module --force purge
+#module use /usr/local/software/jureca/OtherStages
```
same

pramodk commented on sysconfig/jureca-booster/deploy_simstack.sh:
```diff
@@ -0,0 +1,98 @@
+# Create symbolic link
+#mkdir -p $DEPLOYMENT_HOME/../install && cd $DEPLOYMENT_HOME/../install
+#rm -f latest
+#ln -s $date/install/modules/tcl/linux-centos7-x86_64 latest
```
same

jorblancoa replied:
do you mean to remove everything?

pramodk commented on sysconfig/jureca-booster/deploy_simstack.sh:
```diff
@@ -0,0 +1,98 @@
+module load SciPy-Stack/2019a-Python-3.6.8
+module list
+spack spec -Il neurodamus-hippocampus+coreneuron %intel ^coreneuron$cnrn_variant ^python@3.6.8 ^synapsetool%gcc
+spack install --keep-stage --dirty neurodamus-hippocampus+coreneuron %intel ^coreneuron$cnrn_variant ^python@3.6.8 ^synapsetool%gcc
+spack install --keep-stage --dirty neurodamus-neocortex+coreneuron %intel ^coreneuron$cnrn_variant ^python@3.6.8 ^synapsetool%gcc
+spack install --keep-stage --dirty neurodamus-mousify+coreneuron %intel ^coreneuron$cnrn_variant ^python@3.6.8 ^synapsetool%gcc
+
+spack spec -I neuron~mpi %intel ^python@3.6.8
+spack install --dirty --keep-stage -v neuron~mpi %intel ^python@3.6.8
+
+spack spec -I py-bluepy%gcc ^python@3.6.8
+spack install --dirty --keep-stage -v py-bluepy%gcc ^python@3.6.8
+
+spack spec -Il py-sonata-network-reduction%gcc ^python@3.6.8 ^zeromq%intel
+spack install --dirty --keep-stage -v py-sonata-network-reduction%gcc ^python@3.6.8 ^zeromq%intel
+
+spack spec -Il py-bluepyopt%gcc ^python@3.6.8 ^zeromq%intel
+spack install --keep-stage --dirty py-bluepyopt%gcc ^python@3.6.8 ^zeromq%intel
```
nitpick : can we script this bit cleaner? simple changes like: 
```
neurodamus_deps="^coreneuron$cnrn_variant ^python@3.6.8 ^synapsetool%gcc"
for nd in neurodamus-hippocampus ...
do
   spack install --keep-stage --dirty $nd %intel $neurodamus_deps
done
```

Commit:
Address review comments
 * Simplify deployment scripts
 * Add new changes from https://github.com/BlueBrain/spack/pull/759
   - Increase py-pandas version dependency to 0.25 on py-sonata-network-reduction

Commit:
Keep neurodamus-core 2.9.2 as preferred version

Commit:
Fixes for external systems (Jureca & Piz-Daint deployment)
 * Add deployment scripts
 * Update packages.yaml and modules.yaml
 * Update py-sonata-network-reduction package and dependencies
 * Add latest py-jupyter-core and py-jupyter-client from upstream
 * Install zeromq with %intel to avoid issue when running py-bluepyopt (see https://github.com/zeromq/libzmq/issues/3313)

pramodk committed:
Merge branch 'develop' into pr/juelich-update
